sympy/assumptions/smtask.py

```diff
@@ -0,0 +1,191 @@
+"""
+Module to evaluate the proposition with assumptions using SMT algorithm.
+Can handle relational propositions and assumptions.
+"""
+
+from sympy.core.relational import Eq, Ne, Gt, Lt, Ge, Le
+from sympy.assumptions.ask import ask, Q
+from sympy.logic.boolalg import And, Or, Not
+from sympy.assumptions.assume import global_assumptions, AppliedPredicate
+from sympy.printing.smtlib import smtlib_code
+import z3
```

Comment:
We need to make z3 a proper optional dependency. See how this is done in other files (the use in `satisfiable` is a good model). It should also be added here https://github.com/sympy/sympy/blob/master/doc/src/contributing/dependencies.md